src/x509_cert_helper.cpp

```diff
@@ -0,0 +1,123 @@
+#include "mls/x509_cert_helper.h"
+#include "openssl/asn1.h"
+#include "openssl/bio.h"
+#include "openssl/err.h"
+#include "openssl/objects.h"
+#include "openssl/pem.h"
+#include "openssl/rand.h"
+#include <openssl/x509.h>
+#include <openssl/x509v3.h>
+
+namespace mls {
+
+///
+/// OpenSSL X509 Certificate
+///
+
+using EVP_PKEY_ptr = std::unique_ptr<EVP_PKEY, decltype(&::EVP_PKEY_free)>;
+using X509_ptr = std::unique_ptr<X509, decltype(&::X509_free)>;
+
+static std::runtime_error
+openssl_error()
+{
+	uint64_t code = ERR_get_error();
+	return std::runtime_error(ERR_error_string(code, nullptr));
+}
+
+
+struct OpenSSLCertificate : public X509Certificate {
+	explicit OpenSSLCertificate(bytes cert_in) {
+		cert_raw = std::move(cert_in);
+		const unsigned char *buf = cert_raw.data();
+		X509 *tmp = d2i_X509(nullptr, &buf, cert_raw.size());
```

Comment:
1. Always use `auto` for declarations.  Here and on the previous line `const auto*`
2. You should immediately wrap this in a smart pointer, `make_typed_unique(...)`.  If you need to pass it, you can then use `.release()`.

However, I notice that `typed_unique_ptr` is up in the `hpke` library now.  Which might point to a need to move this code up there as well.